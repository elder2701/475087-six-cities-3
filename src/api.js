import axios from "axios";

const Error = {

};

export const createAPI = (onUnauthorized) => {
  const api = axios.create({
    baseURL: `https://htmlacademy-react-3.appspot.com/six-cities`,
    timeount: 5000,
    withCredentials: true
  });

  const onSuccess = (response) => {
    return response;
  };

  const onFail = (err) => {
    const {response} = err;

    if (response.status === Error.UNAUTHORIZED) {
      onUnauthorized();
      throw err;
    }
    throw err;
  };
  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
