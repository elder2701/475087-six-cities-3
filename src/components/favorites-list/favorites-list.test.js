import React from "react";
import renderer from "react-test-renderer";
import FavoritesList from "./favorites-list.jsx";
import history from "../../history.js";
import {Router} from "react-router-dom";
import {Provider} from "react-redux";
import cofigureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import {createAPI} from "../../api.js";
import thunk from "redux-thunk";

const api = createAPI(() => {});

const mockStore = cofigureStore([thunk.withExtraArgument(api)]);

it(`<FavoritesList /> sould be render`, () => {
  jest.mock(`../../reducer/operation/operation.js`, () => ({
    OperationFavorites: {
      loadFavorites: jest.fn(),
      reserFavorites: jest.fn()
    }
  }));
  const store = mockStore({
    [NameSpace.FAVORITE]: {
      favorites: {
        Amsterdam: {
          city: {name: `Amsterdam`},
          offers: [
            {
              isPremium: false,
              previewImage: ``,
              price: 12,
              rating: 1,
              id: 1,
              title: ``,
              type: ``
            }
          ]
        }
      }
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <FavoritesList/>
          </Router>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
