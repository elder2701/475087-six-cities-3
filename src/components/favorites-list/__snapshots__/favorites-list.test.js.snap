// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesList /> sould be render 1`] = `
<div
  className="page__favorites-container container"
>
  <section
    className="favorites"
  >
    <h1
      className="favorites__title"
    >
      Saved listing
    </h1>
    <ul
      className="favorites__list"
    >
      <li
        className="favorites__locations-items"
      >
        <div
          className="favorites__locations locations locations--current"
        >
          <div
            className="locations__item"
          >
            <a
              className="locations__item-link"
              href="/"
              onClick={[Function]}
            >
              <span>
                Amsterdam
              </span>
            </a>
          </div>
        </div>
        <div
          className="favorites__places"
        >
          <article
            className="favorites__card place-card"
          >
            <div
              className="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="110"
                  src=""
                  width="150"
                />
              </a>
            </div>
            <div
              className="favorites__card-info place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    633
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                >
                  The house among olive 
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                house
              </p>
            </div>
          </article>
        </div>
      </li>
      <li
        className="favorites__locations-items"
      >
        <div
          className="favorites__locations locations locations--current"
        >
          <div
            className="locations__item"
          >
            <a
              className="locations__item-link"
              href="/"
              onClick={[Function]}
            >
              <span>
                Brussels
              </span>
            </a>
          </div>
        </div>
        <div
          className="favorites__places"
        >
          <article
            className="favorites__card place-card"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="110"
                  src=""
                  width="150"
                />
              </a>
            </div>
            <div
              className="favorites__card-info place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    274
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                >
                  The Joshua Tree House
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                room
              </p>
            </div>
          </article>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
